### PR TITLE
Add bundle URL annotations to each application when deploying a bundle

### PR DESCRIPTION
**Pull request by hatched**
So that the UI can know which applications were deployed from bundles, this branch adds the `bundle-url` annotation to each application deployed that way.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8297/

File: jujugui/static/gui/src/app/init/bundle-importer.js [hunk 1]
```diff
@@ -106,9 +110,49 @@ class BundleImporter {
...
     this.importBundleDryRun(changes);
   }
 
+  /**
+    Adds the bundle-url annotation to the changeset.
+    @param {String} bundleURL THe bundle url to save to the annotations.
+    @param {Array} changes The bundle changeset.
+  */
+  _addBundleURLAnnotation(bundleURL, changes) {
+    // For each different application records and add an annotation for each one.
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/bundle-importer.js]: This sentence seems a bit weird.

File: jujugui/static/gui/src/app/init/bundle-importer.js [hunk 2]
```diff
@@ -106,9 +110,49 @@ class BundleImporter {
...
       return;
     }
+    // Add the bundle id annotation record.
+    changes = this._addBundleURLAnnotation(bundleURL, changes);
     this.importBundleDryRun(changes);
   }
 
+  /**
+    Adds the bundle-url annotation to the changeset.
+    @param {String} bundleURL THe bundle url to save to the annotations.
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/bundle-importer.js]: URL

**Comment by mitechie** [comment]
Can we go ahead and split/store two annotations in there. One that's just the bundle name and one that's the bundle-url for full reference? It'll mean the front end can do a better job of just grabbing the name field and grouping stuff later.

**Comment by mitechie** [comment]
When using a local bundle we should add an incrementor to the bundle-url so that it's bundle-0, bundle-1, etc so that doing it with more than one bundle will not group things accidently. 

I also think using local: or something would be good to call out.

File: jujugui/static/gui/src/app/init/bundle-importer.js (hunk 2 above)
Comment (hatched) [reply jujugui/static/gui/src/app/init/bundle-importer.js]: done

File: jujugui/static/gui/src/app/init/bundle-importer.js (hunk 1 above)
Comment (hatched) [reply jujugui/static/gui/src/app/init/bundle-importer.js]: Yeah that is a lot weird, rewritten.

**Comment by hatched** [comment]
After an offline discussion with @mitechie we've decided to leave the `bundle-url` annotation as the applications should be grouped by the url and not some inferred name to eliminate grouping issues.

The local bundle file name should be incremented because the GUI will duplicate all applications instead of replacing/upgrading existing ones.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8311/

**Pull request merged**